data/crac-io/crac-io-cne/src/main/java/com/farao_community/farao/data/crac_io_cne/CneFiller.java

```diff
@@ -0,0 +1,395 @@
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.farao_community.farao.data.crac_io_cne;
+
+import com.farao_community.farao.commons.FaraoException;
+import com.farao_community.farao.data.crac_api.*;
+import com.farao_community.farao.data.crac_result_extensions.*;
+import com.powsybl.iidm.network.Branch;
+import com.powsybl.iidm.network.Network;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.xml.datatype.DatatypeConfigurationException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.farao_community.farao.data.crac_io_cne.CneClassCreator.*;
+import static com.farao_community.farao.data.crac_io_cne.CneUtil.*;
+import static com.farao_community.farao.data.crac_io_cne.CneConstants.*;
+
+/**
+ * Fills the classes that constitute the CNE file structure
+ *
+ * @author Viktor Terrier {@literal <viktor.terrier at rte-france.com>}
+ */
+public final class CneFiller {
+
+    private static CriticalNetworkElementMarketDocument cne = new CriticalNetworkElementMarketDocument();
+    private static List<Instant> instants;
+    private static Map<State, ConstraintSeries> constraintSeriesMapB57 = new HashMap<>();
+    private static Map<State, ConstraintSeries> constraintSeriesMapB56 = new HashMap<>();
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CneFiller.class);
+
+    private CneFiller() { }
+
+    public static CriticalNetworkElementMarketDocument getCne() {
+        return cne;
+    }
+
+    /*****************
+     GENERAL METHODS
+     *****************/
+    // Main method
+    public static void generate(Crac crac, Network network, Unit chosenExportUnit) {
+
+        if (!crac.isSynchronized()) {
+            crac.synchronize(network);
+        }
+
+        // sort the instants in order to determine which one is preventive, after outage, after auto RA and after CRA
+        instants = crac.getInstants().stream().sorted(Comparator.comparing(Instant::getSeconds)).collect(Collectors.toList());
+
+        fillHeader(crac.getNetworkDate());
+        addTimeSeriesToCne(crac.getNetworkDate());
+        Point point = cne.getTimeSeries().get(0).getPeriod().get(0).getPoint().get(0);
+
+        if (crac.getExtension(CracResultExtension.class) != null && crac.getExtension(ResultVariantManager.class).getVariants() != null) { // Computation ended
+
+            CracResultExtension cracExtension = crac.getExtension(CracResultExtension.class);
+
+            // TODO: store the information on preOptim/postOptim Variant in the ResultVariantManager
+            String preOptimVariantId;
+            String postOptimVariantId;
+
+            // define preOptimVariant and postOptimVariant
+            List<String> variants = new ArrayList<>(crac.getExtension(ResultVariantManager.class).getVariants());
```

Comment:
I don't think so, because nothing guarantees that there are only two variants. For instance, I've seen cases with only one variant. I don't know if there are other cases with 3 or more variants, but this possibility is taken into account by my code.